vac/raw/mix.md

```diff
@@ -0,0 +1,658 @@
+---
+title: MIX
+name: Mix
+status: raw
+category: Standards Track
+tags:
+editor: Akshaya Mani <akshaya@status.im>
+contributors:
+---
+
+## Abstract
+
+This document specifies the Mix protocol, a custom protocol within the
+[libp2p](https://libp2p.io) framework designed to enable anonymous communication
+in peer-to-peer networks. The Mix protocol allows libp2p nodes to send messages
+without revealing the sender's identity to intermediary nodes or the recipient.
+It achieves this by using the Sphinx packet format, which encrypts and routes
+messages through a series of nodes (mix nodes) before reaching the recipient.
+
+Key features of the protocol include:
+
+i. Path selection for choosing a random route through the network via multiple
+mix nodes.\
+ii. Sphinx packet construction and processing, providing cryptographic
+guarantees of anonymity and security.\
+iii. Proof of Work mechanism to prevent spamming of the mix network.\
```

Comment:
Yes, we can use a different method of spam prevention, such as RLN. However, given the need for a decentralized, simple, and independent spam prevention mechanism in our custom mix protocol, PoW is the preferred choice.

i. PoW does not require a trusted setup or continuous management. It operates independently without relying on any central or semi-trusted authority. 
ii. It does not have dependencies on external systems or blockchains, avoiding the overhead and potential points of failure associated with such integrations. This ensures that our mix protocol remains self-contained and robust.
iii. It provides a straightforward security model by requiring computational effort to send messages. It is simple to implement and understand, making it easier to integrate into our mix protocol without introducing additional complexity.
iv. When implemented properly, PoW scales well. The exit nodes can efficiently verify a large number of incoming messages without significant performance degradation.

Reply:
totally understand now, but in the event that there is a malicious actor with large compute capacity, will the network be able to handle the load?
This is what killed whisper

Reply:
This is a valid question! Our Mix Protocol improves on Whisper by centralizing PoW verification to exit nodes rather than every node along the message path. This reduces the computational load to a linear relationship between the number of messages and verifications, rather than the multiplicative burden seen in Whisper.

While this design helps mitigate computational strain, exit nodes are still vulnerable to targeted attacks. To counter this, PoW difficulty can be dynamically adjusted based on network load, with updated levels communicated via ENRs. Senders can then perform discovery to obtain the current ENR of the chosen exit node(s), ensuring they use the appropriate PoW difficulty. This would help the network adapt to varying conditions. This approach enhances resistance to spam, though it is worth noting that similar vulnerabilities are faced by other anonymity networks as well.

Despite these limitations, PoW is a practical choice due to its decentralized, simple, and independent nature. While alternative mechanisms might offer more robustness, they require trust or blockchain dependencies. Thus, PoW offers a pragmatic solution for initial deployment and can be refined over time to provide more resistance against high-compute capacity attacks.

Reply:
Using Verifiable Delay Functions (VDFs) as an alternative to Proof of Work (PoW) is an interesting idea. VDFs require sequential computation, which eliminates the parallel computing advantages of users with high computational power, creating a more level playing field. They also provide efficient verification and deterministic output, which can enhance the protocol's performance.
However, implementing VDFs may introduce additional complexity and could require a trusted setup, raising new security considerations. While VDFs are promising for future versions of the Mix protocol, we should carefully evaluate their maturity and practical applications before deciding to integrate them.